.github/workflows/tekton_bundle_push.yaml

```diff
@@ -0,0 +1,117 @@
+---
+name: Tekton Bundle Push
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches: ['main']
+  workflow_dispatch:
+env:
+  IMAGE_REGISTRY: quay.io
+  IMAGE_NAMESPACE: ${{ secrets.QUAY_NAMESPACE }}
+  REGISTRY_USER: ${{ secrets.QUAY_ROBOT_USER }}
+  REGISTRY_PASSWORD: ${{ secrets.QUAY_ROBOT_TOKEN }}
+  API_TOKEN: ${{ secrets.QUAY_API_TOKEN }}
+jobs:
+  run-pipeline:
+    name: Tekton Bundle
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2  # using git diff-tree requires fetch depth=2
+      - run: |
+          mkdir -vp $HOME/.kube || true
+          cat <<-EOF > $HOME/.kube/config
+          apiVersion: v1
+          kind: Config
+          clusters:
+          - cluster:
+              server: _
+            name: _
+          contexts:
+          - context:
+              cluster: _
+            name: _
+          current-context: _
+          EOF
+      - uses: jerop/tkn@v0.1.0
+      - uses: redhat-actions/podman-login@v1
+        with:
+          username: ${{ env.REGISTRY_USER }}
+          password: ${{ env.REGISTRY_PASSWORD }}
+          registry: ${{ env.IMAGE_REGISTRY }}
+      - run: |
+          printf 'Gathering new/modified bundle definitions.\n'
+
+          # Gather new/modified bundle definitions.
+          declare -A bundle_dirs  # associative array to hash bundle dirs in key space.
+          while read file
```

Comment:
Yes, that's exactly the way it works right now, but I recognize it might be convoluted.
So it creates a bash associative array, and anytime the loop finds a changed file, the directory that contains the changed file is added to the array. 

Later on the array of changed directories is looped, and each yaml definition file is added to the bundle.